object_detection/distributed_training/setup.md

```diff
@@ -0,0 +1,51 @@
+## Setup Kubeflow
+### Requirements
+
+ - Kubernetes cluster
+ - Access to a working `kubectl` (Kubernetes CLI)
+ - Ksonnet CLI: [ks](https://ksonnet.io/)
+
+### Setup
+Refer to the [user guide](https://github.com/kubeflow/kubeflow/blob/master/user_guide.md) for instructions on how to setup kubeflow on your kubernetes cluster. Specifically, look at section [deploy kubeflow](https://github.com/kubeflow/kubeflow/blob/master/user_guide.md#deploy-kubeflow). 
+For this example we will be using ks `nocloud` environment. If you plan to use `cloud` ks environment, please make sure you follow the proper instructions in the kubeflow user guide.
+
+After completing the steps in the kubeflow user guide you will have the following:
+- A ksonnet app directory called `my-kubeflow` 
+- A new namespace in you K8s cluster called `kubeflow`
+- The following pods in your kubernetes cluster in the `kubeflow` namespace:
+```
+kubectl -n kubeflow get pods
+NAME                              READY     STATUS    RESTARTS   AGE
+ambassador-7987df44b9-4pht8       2/2       Running   0          1m
+ambassador-7987df44b9-dh5h6       2/2       Running   0          1m
+ambassador-7987df44b9-qrgsm       2/2       Running   0          1m
+tf-hub-0                          1/1       Running   0          1m
+tf-job-operator-78757955b-qkg7s   1/1       Running   0          1m
+```
+## Preparing the training data
```

Comment:
I don't think that this is a good solution to write data to k8s worker nodes directly. When using k8s, we should not have to deal with nodes directly. There are a lot of problems associated with dealing with nodes directly. For example - we do not know which nodes will be scheduled for the tfjob worker pods. How do you cleanup this data? We might not even have ssh access to the nodes.

There are many different ways to do this. One way could be that you can create a preprocessing job which requests a PersistentVolumeClaim and stores data after preprocessing in the PVC.

The PV can then be mounted in `ReadOnlyMany` mode by the TFJob workers and use it for training.

Reply:
Thanks for the feedback @ankushagarwal !!
I based the example on a very small cluster (2 nodes) so that's the reason behind those steps. I completely agree with you, this is a very specific case. I'll change the tutorial to use a more general way of getting and using the data.